src/ddspdrum/module.py

```diff
@@ -87,21 +178,11 @@ def __init__(
                                 exponential.
         """
         super().__init__(sample_rate=sample_rate, control_rate=control_rate)
-
-        assert alpha >= 0
-        self.alpha = alpha
-
-        assert s >= 0 and s <= 1
-        self.s = s
-
-        assert a >= 0
-        self.a = a
-
-        assert d >= 0
-        self.d = d
-
-        assert r >= 0
-        self.r = r
+        self.add_parameter("attack", a, 0, 20, scale=0.5)
+        self.add_parameter("decay", d, 0, 20, scale=0.5)
+        self.add_parameter("sustain", s, 0, 1)
+        self.add_parameter("release", r, 0, 20, scale=0.5)
+        self.add_parameter("alpha", alpha, 0, 10)
```

Comment:
Each SynthModule defines named parameters with an initial value, minimum, maximum, and scale. Scale helps in the conversion to and from 0 to 1 range